src/test/java/org/jenkinsci/plugins/github_branch_source/PullRequestSCMRevisionTest.java

```diff
@@ -0,0 +1,289 @@
+        assertThat(abort.getMessage(), containsString("Unknown merge state"));
+    }
+
+    @Test
+    public void createMergewithNotMergeableRevision() throws Exception {
+        PullRequestSCMRevision prRevision = new PullRequestSCMRevision(
+            prMerge, "master-revision", "pr-branch-revision", PullRequestSCMRevision.NOT_MERGEABLE_HASH);
+        assertThat(prRevision.toString(), is("pr-branch-revision+master-revision (NOT_MERGEABLE)"));
+
+        // equality
+        assertThat(prRevision,
+            not(is(new PullRequestSCMRevision(prMerge, "master-revision", "pr-branch-revision", null))));
+        assertThat(prRevision,
+            not(is(new PullRequestSCMRevision(prMerge, "master-revision", "pr-branch-revision", ""))));
+        assertThat(prRevision,
+            not(is(new PullRequestSCMRevision(prMerge, "master-revision", "pr-branch-revision", "pr-merge-revision"))));
+
+        assertThat(prRevision,
+            not(is(new PullRequestSCMRevision(prHead, "master-revision", "pr-branch-revision", null))));
+
+        // validation should fail for this PR.
+        Exception abort = null;
+        try {
+            prRevision.validateMergeHash(repo);
+        } catch (Exception e) {
+            abort = e;
+        }
+        assertThat(abort, instanceOf(AbortException.class));
+        assertThat(abort.getMessage(), containsString("Not mergeable"));
+    }
+
+    @Test
+    public void createMergewithMergeRevision() throws Exception {
+        PullRequestSCMRevision prRevision = new PullRequestSCMRevision(
+            prMerge, "master-revision", "pr-branch-revision", "pr-merge-revision");
+        assertThat(prRevision.toString(), is("pr-branch-revision+master-revision (pr-merge-revision)"));
+
+        // equality
+        assertThat(prRevision,
+            not(is(new PullRequestSCMRevision(prMerge, "master-revision", "pr-branch-revision", null))));
+        assertThat(prRevision,
+            not(is(new PullRequestSCMRevision(prMerge, "master-revision", "pr-branch-revision", ""))));
+        assertThat(prRevision,
+            is(new PullRequestSCMRevision(prMerge, "master-revision", "pr-branch-revision", "pr-merge-revision")));
+
+        assertThat(prRevision,
+            not(is(new PullRequestSCMRevision(prHead, "master-revision", "pr-branch-revision", "pr-merge-revision"))));
+    }
+
+    @Test
+    public void createMergewithMergeRevision_validation_valid() throws Exception {
+        PullRequestSCMRevision prRevision = new PullRequestSCMRevision(
+            prMerge,
+            "8f1314fc3c8284d8c6d5886d473db98f2126071c",
+            "c0e024f89969b976da165eecaa71e09dc60c3da1",
+            "38814ca33833ff5583624c29f305be9133f27a40");
+
+        assertThat(prRevision.toString(), is("c0e024f89969b976da165eecaa71e09dc60c3da1+8f1314fc3c8284d8c6d5886d473db98f2126071c (38814ca33833ff5583624c29f305be9133f27a40)"));
+        prRevision.validateMergeHash(repo);
```

Comment:
```suggestion
        try {
            prRevision.validateMergeHash(repo);
        } catch (AbortException e) {
            fail("Validation should succeed, but: " + e.getMessage())
        }
```

If the validation fails the way it is written it will be reported as an error instead of a failure